tests/lib/tools/nested-state

```diff
@@ -0,0 +1,110 @@
+#!/bin/bash -e
+
+show_help() {
+    echo "usage: prepare-env"
+    echo "       cleanup-env"
+    echo "       create-core-vm"
+    echo "       start-core-vm"
+    echo "       create-classic-vm"
+    echo "       start-classic-vm"
+    echo "       force-start-vm"
```

Comment:
What is the difference between starting and force-starting a virtual machine?

Reply:
I think it makes sense to update the command names, at least, then the internal nested.sh functions, separately. One advantage of the tool vs library approach is that we can refactor things more easily and design a smaller API surface than with a direct library.

If this is rare we could also add an option for it, like `nested-state start-vm --existing`. I'm also interested in reducing the necessity for a particular test to know how `nested-state` works internally, so that this is not a leaky abstraction.